common/src/time_trigger.cpp

```diff
@@ -125,16 +125,18 @@ pcl::TimeTrigger::thread_function ()
   while (true)
   {
     double time = getTime ();
-    boost::unique_lock<boost::mutex> lock (condition_mutex_);
+    std::unique_lock<std::mutex> lock (condition_mutex_);
     if(quit_)
       break;
     if (!running_)
       condition_.wait (lock); // wait util start is called or destructor is called
     else
     {
+      using namespace std::chrono_literals;
+
       callbacks_();
       double rest = interval_ + time - getTime ();
-      condition_.timed_wait (lock, boost::posix_time::microseconds (static_cast<int64_t> ((rest * 1000000))));
+      condition_.wait_for (lock, rest * 1000000us);
```

Comment:
Why not `1s` instead of `1000000us`?

Reply:
I was scared slightly concerned that it might use some internal integer representation than it would mess up the product with `rest`. After thinking about it with a refreshed mind I realize this concern doesn't make much sense. double * integer is likely to get promoted to double anyway so I will change it.

Side note: if you have time, patience and willingness, consider the reviewing PRs from both me and Sergey over the next week. We could benefit for an extra pair of eyes.